### PR TITLE
sql: Default changes from TIDB master

### PR DESCRIPTION
**Pull request by morgo**
These changes AFAIK are in master only, and have not been backported to TiDB 2.1:

- The default charset is now utf8mb4 https://github.com/pingcap/tidb/pull/7965  (simplifying explanation, since TiDB utf8 is like MySQL utf8mb4).
- foreign key checks = 0  https://github.com/pingcap/tidb/pull/8358
- the sql mode will be the same as mysql  https://github.com/pingcap/parser/pull/45

### CODE REVIEW

File: sql/character-set-configuration.md [hunk 1]
```diff
@@ -6,6 +6,6 @@ category: user guide
 
 # Character Set Configuration
 
-Currently, TiDB only supports the `utf8` character set, which is the equivalent to `utf8mb4` in MySQL. Since MySQL 5.7 defaults to `latin1`, this difference is documented under [default differences](../sql/mysql-compatibility.md#default-differences) between TiDB and MySQL.
+Currently, TiDB only supports the `utf8mb4` character set, and its alias `utf8`. Since MySQL 5.7 defaults to `latin1`, this difference is documented under [default differences](../sql/mysql-compatibility.md#default-differences) between TiDB and MySQL.
```
Comment (gregwebs) [inline sql/character-set-configuration.md]: Not your copy, but I would get rid of "Currently".
"and its alias `utf8`" should be changed: in MySQL they are not aliases. I would probably state:
TiDB only supports the `utf8mb4` character set. `utf8` is treated as an alias of `utf8mb4` and is accepted as the recorded character set in some instances. However, in general TiDB will ignore other character sets and record utf8mb4 as the character set being used.

Comment (morgo) [reply sql/character-set-configuration.md]: Agree, I'll change it to improve readability.

**Comment by morgo** [comment]
@queenypingcap This PR is ready to merge when the docs is ready for master-branch-only content.

**Comment by lilin90** [comment]
@morgo Does this change only apply to TiDB 3.0? If yes, we can merge this PR add a `v3.0` label.

File: sql/character-set-support.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ category: user guide
 
 A character set is a set of symbols and encodings. A collation is a set of rules for comparing characters in a character set.
 
-Currently, TiDB supports the following character sets:
+For **compatibility reasons** TiDB lists several character sets as available, but will always store data in the `utf8mb4` character set.
```
Comment (lilin90) [inline sql/character-set-support.md]:
```suggestion
For **compatibility reasons**, TiDB lists several character sets as available, but will always store data in the `utf8mb4` character set.
```

**Comment by morgo** [comment]
Let's DNM the 3.0 PRs until after DITA and version split.

**Comment by morgo** [comment]
Superseded by https://github.com/pingcap/docs/pull/1253

**Pull request closed without merging**